{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## How distant is the nucleotide sequence of SARS coronavirus and SARS-CoV-2 ?\n",
    "\n",
    "This week, I want to compare SARS coronavirus to SARS-CoV-2.\n",
    "\n",
    "I gathered my data from the [NCBI Nucleotide database](https://www.ncbi.nlm.nih.gov/nuccore)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Here is my query used to find SARS coronavirus sequences\n",
    "https://www.ncbi.nlm.nih.gov/nuccore/?term=SARS+coronavirus\n",
    "\n",
    "### Here is my query used to find SARS-CoV-2 sequences\n",
    "https://www.ncbi.nlm.nih.gov/nuccore/?term=SARS-CoV-2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "\n",
    "# The id's for SARS coronavirus sequences\n",
    "accessions_sars_cov = [\"DQ497008\", \"FJ882962\", \"FJ882961\", \"FJ882959\", \"FJ882958\", \"FJ882957\", \" FJ882953\",\n",
    "                      \"FJ882952\", \"FJ882951\", \"FJ882948\", \"FJ882945\", \"FJ882943\", \"FJ882942\", \"AY291451\"]\n",
    "\n",
    "# The id's for SARS-CoV-2 sequences\n",
    "accessions_sars_cov_2 = [\"MT262896\", \"MT262897\", \"MT262898\", \"MT262899\", \"MT262900\", \"MT262901\", \"MT262902\",\n",
    "                      \"MT262903\", \"MT262904\", \"MT262905\", \"MT262906\", \"MT262907\", \"MT262908\", \"MT262909\"]\n",
    "\n",
    "Entrez.api_key = \"9149ff954f582d0ad38507b8cd33140b1b08\"\n",
    "Entrez.email = 'skurdogh@calpoly.edu'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Reading in SARS-CoV-2 sequences using Entrez module"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "handle_2 = Entrez.efetch(db=\"nucleotide\", id=accessions_sars_cov_2, rettype=\"fasta\", retmode=\"xml\")\n",
    "record_2 = Entrez.read(handle_2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Reading in SARS coronavirus sequences using Entrez module"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "handle_1 = Entrez.efetch(db=\"nucleotide\", id=accessions_sars_cov, rettype=\"fasta\", retmode=\"xml\")\n",
    "record_1 = Entrez.read(handle_1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "get_data = []\n",
    "for r in record_1:\n",
    "    d = {\"headline\": r[\"TSeq_defline\"], \"accession\": r[\"TSeq_accver\"], \"sequence\": r[\"TSeq_sequence\"]}\n",
    "    get_data.append(d)\n",
    "    \n",
    "sars_cov_df = pd.DataFrame(get_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>headline</th>\n",
       "      <th>accession</th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SARS coronavirus strain MA-15, complete genome</td>\n",
       "      <td>DQ497008.1</td>\n",
       "      <td>ATATTAGGTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SARS coronavirus MA15 ExoN1 isolate P3pp10, co...</td>\n",
       "      <td>FJ882962.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SARS coronavirus MA15 isolate P3pp5, complete ...</td>\n",
       "      <td>FJ882961.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SARS coronavirus MA15 ExoN1 isolate P3pp6, com...</td>\n",
       "      <td>FJ882959.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SARS coronavirus MA15 isolate P3pp7, complete ...</td>\n",
       "      <td>FJ882958.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SARS coronavirus MA15, complete genome</td>\n",
       "      <td>FJ882957.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SARS coronavirus MA15 ExoN1 isolate P3pp4, com...</td>\n",
       "      <td>FJ882953.1</td>\n",
       "      <td>CTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SARS coronavirus MA15 isolate P3pp4, complete ...</td>\n",
       "      <td>FJ882952.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SARS coronavirus MA15 ExoN1 isolate P3pp3, com...</td>\n",
       "      <td>FJ882951.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SARS coronavirus MA15 isolate P3pp3, complete ...</td>\n",
       "      <td>FJ882948.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SARS coronavirus MA15 isolate P3pp6, complete ...</td>\n",
       "      <td>FJ882945.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SARS coronavirus MA15 ExoN1, complete genome</td>\n",
       "      <td>FJ882943.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SARS coronavirus MA15 ExoN1 isolate P3pp5, com...</td>\n",
       "      <td>FJ882942.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SARS coronavirus TW1, complete genome</td>\n",
       "      <td>AY291451.1</td>\n",
       "      <td>ATATTAGGTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCT...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             headline   accession  \\\n",
       "0      SARS coronavirus strain MA-15, complete genome  DQ497008.1   \n",
       "1   SARS coronavirus MA15 ExoN1 isolate P3pp10, co...  FJ882962.1   \n",
       "2   SARS coronavirus MA15 isolate P3pp5, complete ...  FJ882961.1   \n",
       "3   SARS coronavirus MA15 ExoN1 isolate P3pp6, com...  FJ882959.1   \n",
       "4   SARS coronavirus MA15 isolate P3pp7, complete ...  FJ882958.1   \n",
       "5              SARS coronavirus MA15, complete genome  FJ882957.1   \n",
       "6   SARS coronavirus MA15 ExoN1 isolate P3pp4, com...  FJ882953.1   \n",
       "7   SARS coronavirus MA15 isolate P3pp4, complete ...  FJ882952.1   \n",
       "8   SARS coronavirus MA15 ExoN1 isolate P3pp3, com...  FJ882951.1   \n",
       "9   SARS coronavirus MA15 isolate P3pp3, complete ...  FJ882948.1   \n",
       "10  SARS coronavirus MA15 isolate P3pp6, complete ...  FJ882945.1   \n",
       "11       SARS coronavirus MA15 ExoN1, complete genome  FJ882943.1   \n",
       "12  SARS coronavirus MA15 ExoN1 isolate P3pp5, com...  FJ882942.1   \n",
       "13              SARS coronavirus TW1, complete genome  AY291451.1   \n",
       "\n",
       "                                             sequence  \n",
       "0   ATATTAGGTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCT...  \n",
       "1   GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "2   GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "3   GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "4   GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "5   GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "6   CTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGT...  \n",
       "7   GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "8   GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "9   GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "10  GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "11  GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "12  GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "13  ATATTAGGTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCT...  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sars_cov_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "get_data = []\n",
    "for r in record_2:\n",
    "    d = {\"headline\": r[\"TSeq_defline\"][55:], \"accession\": r[\"TSeq_accver\"], \"sequence\": r[\"TSeq_sequence\"]}\n",
    "    get_data.append(d)\n",
    "    \n",
    "sars_cov_2_df = pd.DataFrame(get_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>headline</th>\n",
       "      <th>accession</th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SARS-CoV-2/human/USA/WA-NH2/2020, complete ge...</td>\n",
       "      <td>MT262896.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SARS-CoV-2/human/USA/WA-NH3/2020, complete ge...</td>\n",
       "      <td>MT262897.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SARS-CoV-2/human/USA/WA-NH4/2020, complete ge...</td>\n",
       "      <td>MT262898.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SARS-CoV-2/human/USA/WA-NH5/2020, complete ge...</td>\n",
       "      <td>MT262899.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SARS-CoV-2/human/USA/WA-NH6/2020, complete ge...</td>\n",
       "      <td>MT262900.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SARS-CoV-2/human/USA/WA-NH7/2020, complete ge...</td>\n",
       "      <td>MT262901.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SARS-CoV-2/human/USA/WA-NH8/2020, complete ge...</td>\n",
       "      <td>MT262902.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SARS-CoV-2/human/USA/WA-NH9/2020, complete ge...</td>\n",
       "      <td>MT262903.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SARS-CoV-2/human/USA/WA-NH10/2020, complete g...</td>\n",
       "      <td>MT262904.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SARS-CoV-2/human/USA/WA-NH11/2020, complete g...</td>\n",
       "      <td>MT262905.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SARS-CoV-2/human/USA/WA-NH12/2020, complete g...</td>\n",
       "      <td>MT262906.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SARS-CoV-2/human/USA/WA-NH14/2020, complete g...</td>\n",
       "      <td>MT262907.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SARS-CoV-2/human/USA/WA-NH17/2020, complete g...</td>\n",
       "      <td>MT262908.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SARS-CoV-2/human/USA/WA-NH18/2020, complete g...</td>\n",
       "      <td>MT262909.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             headline   accession  \\\n",
       "0    SARS-CoV-2/human/USA/WA-NH2/2020, complete ge...  MT262896.1   \n",
       "1    SARS-CoV-2/human/USA/WA-NH3/2020, complete ge...  MT262897.1   \n",
       "2    SARS-CoV-2/human/USA/WA-NH4/2020, complete ge...  MT262898.1   \n",
       "3    SARS-CoV-2/human/USA/WA-NH5/2020, complete ge...  MT262899.1   \n",
       "4    SARS-CoV-2/human/USA/WA-NH6/2020, complete ge...  MT262900.1   \n",
       "5    SARS-CoV-2/human/USA/WA-NH7/2020, complete ge...  MT262901.1   \n",
       "6    SARS-CoV-2/human/USA/WA-NH8/2020, complete ge...  MT262902.1   \n",
       "7    SARS-CoV-2/human/USA/WA-NH9/2020, complete ge...  MT262903.1   \n",
       "8    SARS-CoV-2/human/USA/WA-NH10/2020, complete g...  MT262904.1   \n",
       "9    SARS-CoV-2/human/USA/WA-NH11/2020, complete g...  MT262905.1   \n",
       "10   SARS-CoV-2/human/USA/WA-NH12/2020, complete g...  MT262906.1   \n",
       "11   SARS-CoV-2/human/USA/WA-NH14/2020, complete g...  MT262907.1   \n",
       "12   SARS-CoV-2/human/USA/WA-NH17/2020, complete g...  MT262908.1   \n",
       "13   SARS-CoV-2/human/USA/WA-NH18/2020, complete g...  MT262909.1   \n",
       "\n",
       "                                             sequence  \n",
       "0   ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       "1   ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       "2   ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       "3   ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       "4   ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       "5   ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       "6   ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       "7   ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       "8   ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       "9   ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       "10  ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       "11  ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       "12  ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       "13  ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sars_cov_2_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1. To begin the comparison, I want to know what are the most frequent k-mers in both coronaviruses.\n",
    "### Do they have similar frequent k-mers? What might similar nucleotide patterns say about the relationship between SARS and SARS-CoV-2?\n",
    "\n",
    "The `frequent_words()` method iterates through each nucleotide of a sequence and slices out a pattern of k-length. It then uses a sub routine called `count()` to again iterate through the whole sequence and retrieve the number of times a pattern occurs. Lastly, we retrieve the index of the `max_count` words and go back into our nucleotide sequence at the max_count indices to retrive the frequent patterns."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "\n",
    "def count(text,pattern):\n",
    "    count = 0\n",
    "    len_pattern = len(pattern)\n",
    "    for i in range(len(text) - len_pattern + 1):\n",
    "        substring = text[i:len_pattern + i]\n",
    "        if substring == pattern:\n",
    "            count += 1\n",
    "    return count\n",
    "\n",
    "def frequency_table(text,k):\n",
    "    freq_map = {}\n",
    "    n = len(text)\n",
    "    for i in range(n-k+1):\n",
    "        pattern = text[i:k+i]\n",
    "        try:\n",
    "            freq_map[pattern] += 1\n",
    "        except:\n",
    "            freq_map[pattern] = 1\n",
    "    return freq_map\n",
    "\n",
    "def better_frequent_words(text,k):\n",
    "    frequent_patterns = []\n",
    "    freq_map = frequency_table(text,k)\n",
    "    max_value = max(freq_map.values())\n",
    "    for pattern in freq_map:\n",
    "        if freq_map[pattern] == max_value:\n",
    "            frequent_patterns.append(pattern)\n",
    "    return frequent_patterns,max_value"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get the frequent i-mers in the SARS-CoV-2 sequence and the SARS coronavirus sequence\n",
    "I chose to look at the frequent 10-mers through 15-mers and I store the frequent patterns in a dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "combined_frequent = []\n",
    "for j in range(5, 17):\n",
    "    frequent_i_mers = []\n",
    "    for i in range(len(sars_cov_2_df)):\n",
    "        frequent_patterns, max_val = better_frequent_words(sars_cov_2_df.iloc[i]['sequence'], j)\n",
    "        for f in frequent_patterns:\n",
    "            if f not in frequent_i_mers:\n",
    "                frequent_i_mers.append(f)\n",
    "                \n",
    "    d = {\"i-mer\": j, \"frequent i-mers\": frequent_i_mers}\n",
    "    combined_frequent.append(d)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>i-mer</th>\n",
       "      <th>frequent i-mers</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>[TTGTT]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6</td>\n",
       "      <td>[TTGTTA]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7</td>\n",
       "      <td>[TTGTTAA]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8</td>\n",
       "      <td>[ATGGTGTT]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>[TAAACGAAC]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>10</td>\n",
       "      <td>[CTAAACGAAC, TAATGGTGTT]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>11</td>\n",
       "      <td>[TCTAAACGAAC, TTAAAGTTACA, AATTGTGTTAA, TGGTGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>12</td>\n",
       "      <td>[GTTGATGGTGTT]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>13</td>\n",
       "      <td>[GGCTACTAACAAT, TGATGGTAACAAA, CTTCACACTCAAA, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>14</td>\n",
       "      <td>[ATCAGACAACTACT, TCAGACAACTACTA, CAGACAACTACTA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>15</td>\n",
       "      <td>[ATCAGACAACTACTA, TCAGACAACTACTAT, CAGACAACTAC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>16</td>\n",
       "      <td>[ATCAGACAACTACTAT, TCAGACAACTACTATT, CAATTATTA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    i-mer                                    frequent i-mers\n",
       "0       5                                            [TTGTT]\n",
       "1       6                                           [TTGTTA]\n",
       "2       7                                          [TTGTTAA]\n",
       "3       8                                         [ATGGTGTT]\n",
       "4       9                                        [TAAACGAAC]\n",
       "5      10                           [CTAAACGAAC, TAATGGTGTT]\n",
       "6      11  [TCTAAACGAAC, TTAAAGTTACA, AATTGTGTTAA, TGGTGT...\n",
       "7      12                                     [GTTGATGGTGTT]\n",
       "8      13  [GGCTACTAACAAT, TGATGGTAACAAA, CTTCACACTCAAA, ...\n",
       "9      14  [ATCAGACAACTACT, TCAGACAACTACTA, CAGACAACTACTA...\n",
       "10     15  [ATCAGACAACTACTA, TCAGACAACTACTAT, CAGACAACTAC...\n",
       "11     16  [ATCAGACAACTACTAT, TCAGACAACTACTATT, CAATTATTA..."
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sars_cov_2_frequent = pd.DataFrame(combined_frequent)\n",
    "sars_cov_2_frequent"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "combined_frequent = []\n",
    "for j in range(5, 17):\n",
    "    frequent_i_mers = []\n",
    "    for i in range(len(sars_cov_df)):\n",
    "        frequent_patterns, max_val = better_frequent_words(sars_cov_df.iloc[i]['sequence'], j)\n",
    "        for f in frequent_patterns:\n",
    "            if f not in frequent_i_mers:\n",
    "                frequent_i_mers.append(f)\n",
    "                \n",
    "    d = {\"i-mer\": j, \"frequent i-mers\": frequent_i_mers}\n",
    "    combined_frequent.append(d)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>i-mer</th>\n",
       "      <th>frequent i-mers</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>[TGCTG]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6</td>\n",
       "      <td>[TTGCTG, TGCTGC, TTGATG]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7</td>\n",
       "      <td>[TGCTGCT]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8</td>\n",
       "      <td>[TGTTGACA, TGCTGCTA]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>[TAAACGAAC, ATGTGGCTA, TGATGTTTC]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>10</td>\n",
       "      <td>[CTAAACGAAC, ACTTAGGTGA, TGAGGAAGAA, GAGGAAGAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>11</td>\n",
       "      <td>[TCTAAACGAAC, TGAGGAAGAAG, GAGGAAGAAGA, TGTGTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>12</td>\n",
       "      <td>[TGAGGAAGAAGA, TAAAATGTCTGA]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>13</td>\n",
       "      <td>[AAAAGAAAAAGAC, AAAGAAAAAGACT, AAGAAAAAGACTG, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>14</td>\n",
       "      <td>[AAAAGAAAAAGACT, AAAGAAAAAGACTG, AAGAAAAAGACTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>15</td>\n",
       "      <td>[AAAAGAAAAAGACTG, AAAGAAAAAGACTGA, TTGACATCGTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>16</td>\n",
       "      <td>[AAAAGAAAAAGACTGA]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    i-mer                                    frequent i-mers\n",
       "0       5                                            [TGCTG]\n",
       "1       6                           [TTGCTG, TGCTGC, TTGATG]\n",
       "2       7                                          [TGCTGCT]\n",
       "3       8                               [TGTTGACA, TGCTGCTA]\n",
       "4       9                  [TAAACGAAC, ATGTGGCTA, TGATGTTTC]\n",
       "5      10  [CTAAACGAAC, ACTTAGGTGA, TGAGGAAGAA, GAGGAAGAA...\n",
       "6      11  [TCTAAACGAAC, TGAGGAAGAAG, GAGGAAGAAGA, TGTGTT...\n",
       "7      12                       [TGAGGAAGAAGA, TAAAATGTCTGA]\n",
       "8      13  [AAAAGAAAAAGAC, AAAGAAAAAGACT, AAGAAAAAGACTG, ...\n",
       "9      14  [AAAAGAAAAAGACT, AAAGAAAAAGACTG, AAGAAAAAGACTG...\n",
       "10     15  [AAAAGAAAAAGACTG, AAAGAAAAAGACTGA, TTGACATCGTC...\n",
       "11     16                                 [AAAAGAAAAAGACTGA]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sars_cov_frequent = pd.DataFrame(combined_frequent)\n",
    "sars_cov_frequent"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Now, let's see if they have any of the same frequent i-mers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>i-mer</th>\n",
       "      <th>SARS count of frequent i-mer</th>\n",
       "      <th>SARS-CoV-2 count of frequent i-mer</th>\n",
       "      <th>common i-mer count</th>\n",
       "      <th>common patterns</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>[TAAACGAAC]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>10</td>\n",
       "      <td>33</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>[CTAAACGAAC]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>11</td>\n",
       "      <td>10</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>[TCTAAACGAAC]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>13</td>\n",
       "      <td>32</td>\n",
       "      <td>36</td>\n",
       "      <td>2</td>\n",
       "      <td>[AAACGAACATGAA, AACGAACATGAAA]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>[AAACGAACATGAAA]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>15</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    i-mer  SARS count of frequent i-mer  SARS-CoV-2 count of frequent i-mer  \\\n",
       "0       5                             1                                   1   \n",
       "1       6                             3                                   1   \n",
       "2       7                             1                                   1   \n",
       "3       8                             2                                   1   \n",
       "4       9                             3                                   1   \n",
       "5      10                            33                                   2   \n",
       "6      11                            10                                   7   \n",
       "7      12                             2                                   1   \n",
       "8      13                            32                                  36   \n",
       "9      14                            15                                  16   \n",
       "10     15                             4                                  10   \n",
       "11     16                             1                                   5   \n",
       "\n",
       "    common i-mer count                 common patterns  \n",
       "0                    0                              []  \n",
       "1                    0                              []  \n",
       "2                    0                              []  \n",
       "3                    0                              []  \n",
       "4                    1                     [TAAACGAAC]  \n",
       "5                    1                    [CTAAACGAAC]  \n",
       "6                    1                   [TCTAAACGAAC]  \n",
       "7                    0                              []  \n",
       "8                    2  [AAACGAACATGAA, AACGAACATGAAA]  \n",
       "9                    1                [AAACGAACATGAAA]  \n",
       "10                   0                              []  \n",
       "11                   0                              []  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = []\n",
    "for i in range(len(sars_cov_frequent)):\n",
    "    sars = sars_cov_frequent.iloc[i][\"frequent i-mers\"]\n",
    "    sars_2 = sars_cov_2_frequent.iloc[i][\"frequent i-mers\"]\n",
    "    similar = 0\n",
    "    common_pattern = []\n",
    "    for pattern in sars:\n",
    "        if pattern in sars_2:\n",
    "            common_pattern.append(pattern)\n",
    "            similar += 1\n",
    "    d = {\"i-mer\": sars_cov_frequent.iloc[i][\"i-mer\"], \"SARS count of frequent i-mer\": len(sars), \n",
    "         \"SARS-CoV-2 count of frequent i-mer\": len(sars_2), \"common i-mer count\": similar,\n",
    "        \"common patterns\": common_pattern}\n",
    "    data.append(d)\n",
    "    \n",
    "common_patterns_df = pd.DataFrame(data)\n",
    "common_patterns_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## What Can I Conclude?\n",
    "### It looks like the frequent i-mer patterns in SARS are not likely to be found in SARS-CoV-2. At most, either 1 or 2 frequent i-mers were commob between the two virus sequences. I know the data I used is not super large and diverse."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 2. Now, let's create a phylogenic tree to visually display the relationship between SARS and SARS-CoV-2."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I downloaded a package called [editdistance](https://pypi.org/project/editdistance/0.3.1/#description) to help with build my distance matrix."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "import editdistance"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's stack our two dataframes. This will help with indexing when we build our distance matrix."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession</th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>headline</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus strain MA-15, complete genome</th>\n",
       "      <td>DQ497008.1</td>\n",
       "      <td>ATATTAGGTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 ExoN1 isolate P3pp10, complete genome</th>\n",
       "      <td>FJ882962.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 isolate P3pp5, complete genome</th>\n",
       "      <td>FJ882961.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 ExoN1 isolate P3pp6, complete genome</th>\n",
       "      <td>FJ882959.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 isolate P3pp7, complete genome</th>\n",
       "      <td>FJ882958.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15, complete genome</th>\n",
       "      <td>FJ882957.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 ExoN1 isolate P3pp4, complete genome</th>\n",
       "      <td>FJ882953.1</td>\n",
       "      <td>CTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 isolate P3pp4, complete genome</th>\n",
       "      <td>FJ882952.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 ExoN1 isolate P3pp3, complete genome</th>\n",
       "      <td>FJ882951.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 isolate P3pp3, complete genome</th>\n",
       "      <td>FJ882948.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 isolate P3pp6, complete genome</th>\n",
       "      <td>FJ882945.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 ExoN1, complete genome</th>\n",
       "      <td>FJ882943.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 ExoN1 isolate P3pp5, complete genome</th>\n",
       "      <td>FJ882942.1</td>\n",
       "      <td>GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus TW1, complete genome</th>\n",
       "      <td>AY291451.1</td>\n",
       "      <td>ATATTAGGTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH2/2020, complete genome</th>\n",
       "      <td>MT262896.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH3/2020, complete genome</th>\n",
       "      <td>MT262897.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH4/2020, complete genome</th>\n",
       "      <td>MT262898.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH5/2020, complete genome</th>\n",
       "      <td>MT262899.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH6/2020, complete genome</th>\n",
       "      <td>MT262900.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH7/2020, complete genome</th>\n",
       "      <td>MT262901.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH8/2020, complete genome</th>\n",
       "      <td>MT262902.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH9/2020, complete genome</th>\n",
       "      <td>MT262903.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH10/2020, complete genome</th>\n",
       "      <td>MT262904.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH11/2020, complete genome</th>\n",
       "      <td>MT262905.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH12/2020, complete genome</th>\n",
       "      <td>MT262906.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH14/2020, complete genome</th>\n",
       "      <td>MT262907.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH17/2020, complete genome</th>\n",
       "      <td>MT262908.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH18/2020, complete genome</th>\n",
       "      <td>MT262909.1</td>\n",
       "      <td>ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     accession  \\\n",
       "headline                                                         \n",
       "SARS coronavirus strain MA-15, complete genome      DQ497008.1   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...  FJ882962.1   \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...  FJ882961.1   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...  FJ882959.1   \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...  FJ882958.1   \n",
       "SARS coronavirus MA15, complete genome              FJ882957.1   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...  FJ882953.1   \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...  FJ882952.1   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...  FJ882951.1   \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...  FJ882948.1   \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...  FJ882945.1   \n",
       "SARS coronavirus MA15 ExoN1, complete genome        FJ882943.1   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...  FJ882942.1   \n",
       "SARS coronavirus TW1, complete genome               AY291451.1   \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome  MT262896.1   \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome  MT262897.1   \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome  MT262898.1   \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome  MT262899.1   \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome  MT262900.1   \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome  MT262901.1   \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome  MT262902.1   \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome  MT262903.1   \n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...  MT262904.1   \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...  MT262905.1   \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...  MT262906.1   \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...  MT262907.1   \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...  MT262908.1   \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...  MT262909.1   \n",
       "\n",
       "                                                                                             sequence  \n",
       "headline                                                                                               \n",
       "SARS coronavirus strain MA-15, complete genome      ATATTAGGTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCT...  \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...  GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...  GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...  GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...  GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "SARS coronavirus MA15, complete genome              GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...  CTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGT...  \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...  GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...  GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...  GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...  GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "SARS coronavirus MA15 ExoN1, complete genome        GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...  GTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCTCTTGTAG...  \n",
       "SARS coronavirus TW1, complete genome               ATATTAGGTTTTTACCTACCCAGGAAAAGCCAACCAACCTCGATCT...  \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome  ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome  ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome  ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome  ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome  ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome  ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome  ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome  ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...  ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...  ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...  ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...  ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...  ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...  ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGA...  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_df = pd.concat([sars_cov_df, sars_cov_2_df], axis=0)\n",
    "combined_df = combined_df.set_index('headline')\n",
    "combined_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "distances = {}\n",
    "for i, seqid1 in enumerate(combined_df.index):\n",
    "    distances[seqid1,seqid1]=0\n",
    "    for j in range(i+1,len(combined_df.index)):\n",
    "        seqid2 = combined_df.index[j]\n",
    "        distances[seqid1,seqid2] = editdistance.eval(combined_df.loc[seqid1]['sequence'], combined_df.loc[seqid2]['sequence'])\n",
    "        distances[seqid2,seqid1] = distances[seqid1,seqid2]\n",
    "distances = pd.Series(distances).unstack()\n",
    "distances"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Now, we have our distance matrix!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH10/2020, complete genome</th>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH11/2020, complete genome</th>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH12/2020, complete genome</th>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH14/2020, complete genome</th>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH17/2020, complete genome</th>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH18/2020, complete genome</th>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH2/2020, complete genome</th>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH3/2020, complete genome</th>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH4/2020, complete genome</th>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH5/2020, complete genome</th>\n",
       "      <th>...</th>\n",
       "      <th>SARS coronavirus MA15 ExoN1 isolate P3pp6, complete genome</th>\n",
       "      <th>SARS coronavirus MA15 ExoN1, complete genome</th>\n",
       "      <th>SARS coronavirus MA15 isolate P3pp3, complete genome</th>\n",
       "      <th>SARS coronavirus MA15 isolate P3pp4, complete genome</th>\n",
       "      <th>SARS coronavirus MA15 isolate P3pp5, complete genome</th>\n",
       "      <th>SARS coronavirus MA15 isolate P3pp6, complete genome</th>\n",
       "      <th>SARS coronavirus MA15 isolate P3pp7, complete genome</th>\n",
       "      <th>SARS coronavirus MA15, complete genome</th>\n",
       "      <th>SARS coronavirus TW1, complete genome</th>\n",
       "      <th>SARS coronavirus strain MA-15, complete genome</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH10/2020, complete genome</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>6028</td>\n",
       "      <td>6018</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6014</td>\n",
       "      <td>6014</td>\n",
       "      <td>6015</td>\n",
       "      <td>6011</td>\n",
       "      <td>5993</td>\n",
       "      <td>6001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH11/2020, complete genome</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>6027</td>\n",
       "      <td>6017</td>\n",
       "      <td>6012</td>\n",
       "      <td>6012</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6014</td>\n",
       "      <td>6010</td>\n",
       "      <td>5992</td>\n",
       "      <td>6000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH12/2020, complete genome</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>6028</td>\n",
       "      <td>6018</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6014</td>\n",
       "      <td>6014</td>\n",
       "      <td>6015</td>\n",
       "      <td>6011</td>\n",
       "      <td>5993</td>\n",
       "      <td>6001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH14/2020, complete genome</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>6027</td>\n",
       "      <td>6017</td>\n",
       "      <td>6012</td>\n",
       "      <td>6012</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6014</td>\n",
       "      <td>6010</td>\n",
       "      <td>5992</td>\n",
       "      <td>6000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH17/2020, complete genome</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>6028</td>\n",
       "      <td>6018</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6014</td>\n",
       "      <td>6014</td>\n",
       "      <td>6015</td>\n",
       "      <td>6011</td>\n",
       "      <td>5993</td>\n",
       "      <td>6001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH18/2020, complete genome</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>6028</td>\n",
       "      <td>6018</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6014</td>\n",
       "      <td>6014</td>\n",
       "      <td>6015</td>\n",
       "      <td>6011</td>\n",
       "      <td>5993</td>\n",
       "      <td>6001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH2/2020, complete genome</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>6028</td>\n",
       "      <td>6018</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6014</td>\n",
       "      <td>6014</td>\n",
       "      <td>6015</td>\n",
       "      <td>6011</td>\n",
       "      <td>5993</td>\n",
       "      <td>6001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH3/2020, complete genome</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>6028</td>\n",
       "      <td>6018</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6014</td>\n",
       "      <td>6014</td>\n",
       "      <td>6015</td>\n",
       "      <td>6011</td>\n",
       "      <td>5993</td>\n",
       "      <td>6001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH4/2020, complete genome</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>6028</td>\n",
       "      <td>6018</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6014</td>\n",
       "      <td>6014</td>\n",
       "      <td>6015</td>\n",
       "      <td>6011</td>\n",
       "      <td>5993</td>\n",
       "      <td>6001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH5/2020, complete genome</th>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>6028</td>\n",
       "      <td>6018</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6014</td>\n",
       "      <td>6014</td>\n",
       "      <td>6015</td>\n",
       "      <td>6011</td>\n",
       "      <td>5993</td>\n",
       "      <td>6001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH6/2020, complete genome</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>6027</td>\n",
       "      <td>6017</td>\n",
       "      <td>6012</td>\n",
       "      <td>6012</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6014</td>\n",
       "      <td>6010</td>\n",
       "      <td>5992</td>\n",
       "      <td>6000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH7/2020, complete genome</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>6027</td>\n",
       "      <td>6017</td>\n",
       "      <td>6012</td>\n",
       "      <td>6012</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6014</td>\n",
       "      <td>6010</td>\n",
       "      <td>5992</td>\n",
       "      <td>6000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH8/2020, complete genome</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>6028</td>\n",
       "      <td>6018</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6014</td>\n",
       "      <td>6014</td>\n",
       "      <td>6015</td>\n",
       "      <td>6011</td>\n",
       "      <td>5993</td>\n",
       "      <td>6001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS-CoV-2/human/USA/WA-NH9/2020, complete genome</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>6028</td>\n",
       "      <td>6018</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6014</td>\n",
       "      <td>6014</td>\n",
       "      <td>6015</td>\n",
       "      <td>6011</td>\n",
       "      <td>5993</td>\n",
       "      <td>6001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 ExoN1 isolate P3pp10, complete genome</th>\n",
       "      <td>6027</td>\n",
       "      <td>6026</td>\n",
       "      <td>6027</td>\n",
       "      <td>6026</td>\n",
       "      <td>6027</td>\n",
       "      <td>6027</td>\n",
       "      <td>6027</td>\n",
       "      <td>6027</td>\n",
       "      <td>6027</td>\n",
       "      <td>6027</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>21</td>\n",
       "      <td>25</td>\n",
       "      <td>24</td>\n",
       "      <td>26</td>\n",
       "      <td>25</td>\n",
       "      <td>26</td>\n",
       "      <td>22</td>\n",
       "      <td>58</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 ExoN1 isolate P3pp3, complete genome</th>\n",
       "      <td>6025</td>\n",
       "      <td>6024</td>\n",
       "      <td>6025</td>\n",
       "      <td>6024</td>\n",
       "      <td>6025</td>\n",
       "      <td>6025</td>\n",
       "      <td>6025</td>\n",
       "      <td>6025</td>\n",
       "      <td>6025</td>\n",
       "      <td>6025</td>\n",
       "      <td>...</td>\n",
       "      <td>16</td>\n",
       "      <td>17</td>\n",
       "      <td>21</td>\n",
       "      <td>20</td>\n",
       "      <td>22</td>\n",
       "      <td>21</td>\n",
       "      <td>22</td>\n",
       "      <td>18</td>\n",
       "      <td>54</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 ExoN1 isolate P3pp4, complete genome</th>\n",
       "      <td>6089</td>\n",
       "      <td>6088</td>\n",
       "      <td>6089</td>\n",
       "      <td>6088</td>\n",
       "      <td>6089</td>\n",
       "      <td>6089</td>\n",
       "      <td>6089</td>\n",
       "      <td>6089</td>\n",
       "      <td>6089</td>\n",
       "      <td>6089</td>\n",
       "      <td>...</td>\n",
       "      <td>106</td>\n",
       "      <td>85</td>\n",
       "      <td>91</td>\n",
       "      <td>90</td>\n",
       "      <td>92</td>\n",
       "      <td>91</td>\n",
       "      <td>92</td>\n",
       "      <td>88</td>\n",
       "      <td>124</td>\n",
       "      <td>112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 ExoN1 isolate P3pp5, complete genome</th>\n",
       "      <td>6022</td>\n",
       "      <td>6021</td>\n",
       "      <td>6022</td>\n",
       "      <td>6021</td>\n",
       "      <td>6022</td>\n",
       "      <td>6022</td>\n",
       "      <td>6022</td>\n",
       "      <td>6022</td>\n",
       "      <td>6022</td>\n",
       "      <td>6022</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>23</td>\n",
       "      <td>27</td>\n",
       "      <td>26</td>\n",
       "      <td>28</td>\n",
       "      <td>27</td>\n",
       "      <td>28</td>\n",
       "      <td>24</td>\n",
       "      <td>60</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 ExoN1 isolate P3pp6, complete genome</th>\n",
       "      <td>6028</td>\n",
       "      <td>6027</td>\n",
       "      <td>6028</td>\n",
       "      <td>6027</td>\n",
       "      <td>6028</td>\n",
       "      <td>6028</td>\n",
       "      <td>6028</td>\n",
       "      <td>6028</td>\n",
       "      <td>6028</td>\n",
       "      <td>6028</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>29</td>\n",
       "      <td>28</td>\n",
       "      <td>30</td>\n",
       "      <td>29</td>\n",
       "      <td>30</td>\n",
       "      <td>26</td>\n",
       "      <td>62</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 ExoN1, complete genome</th>\n",
       "      <td>6018</td>\n",
       "      <td>6017</td>\n",
       "      <td>6018</td>\n",
       "      <td>6017</td>\n",
       "      <td>6018</td>\n",
       "      <td>6018</td>\n",
       "      <td>6018</td>\n",
       "      <td>6018</td>\n",
       "      <td>6018</td>\n",
       "      <td>6018</td>\n",
       "      <td>...</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>11</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td>43</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 isolate P3pp3, complete genome</th>\n",
       "      <td>6013</td>\n",
       "      <td>6012</td>\n",
       "      <td>6013</td>\n",
       "      <td>6012</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>...</td>\n",
       "      <td>29</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>39</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 isolate P3pp4, complete genome</th>\n",
       "      <td>6013</td>\n",
       "      <td>6012</td>\n",
       "      <td>6013</td>\n",
       "      <td>6012</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>6013</td>\n",
       "      <td>...</td>\n",
       "      <td>28</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>38</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 isolate P3pp5, complete genome</th>\n",
       "      <td>6014</td>\n",
       "      <td>6013</td>\n",
       "      <td>6014</td>\n",
       "      <td>6013</td>\n",
       "      <td>6014</td>\n",
       "      <td>6014</td>\n",
       "      <td>6014</td>\n",
       "      <td>6014</td>\n",
       "      <td>6014</td>\n",
       "      <td>6014</td>\n",
       "      <td>...</td>\n",
       "      <td>30</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>40</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 isolate P3pp6, complete genome</th>\n",
       "      <td>6014</td>\n",
       "      <td>6013</td>\n",
       "      <td>6014</td>\n",
       "      <td>6013</td>\n",
       "      <td>6014</td>\n",
       "      <td>6014</td>\n",
       "      <td>6014</td>\n",
       "      <td>6014</td>\n",
       "      <td>6014</td>\n",
       "      <td>6014</td>\n",
       "      <td>...</td>\n",
       "      <td>29</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>39</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15 isolate P3pp7, complete genome</th>\n",
       "      <td>6015</td>\n",
       "      <td>6014</td>\n",
       "      <td>6015</td>\n",
       "      <td>6014</td>\n",
       "      <td>6015</td>\n",
       "      <td>6015</td>\n",
       "      <td>6015</td>\n",
       "      <td>6015</td>\n",
       "      <td>6015</td>\n",
       "      <td>6015</td>\n",
       "      <td>...</td>\n",
       "      <td>30</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>40</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus MA15, complete genome</th>\n",
       "      <td>6011</td>\n",
       "      <td>6010</td>\n",
       "      <td>6011</td>\n",
       "      <td>6010</td>\n",
       "      <td>6011</td>\n",
       "      <td>6011</td>\n",
       "      <td>6011</td>\n",
       "      <td>6011</td>\n",
       "      <td>6011</td>\n",
       "      <td>6011</td>\n",
       "      <td>...</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>36</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus TW1, complete genome</th>\n",
       "      <td>5993</td>\n",
       "      <td>5992</td>\n",
       "      <td>5993</td>\n",
       "      <td>5992</td>\n",
       "      <td>5993</td>\n",
       "      <td>5993</td>\n",
       "      <td>5993</td>\n",
       "      <td>5993</td>\n",
       "      <td>5993</td>\n",
       "      <td>5993</td>\n",
       "      <td>...</td>\n",
       "      <td>62</td>\n",
       "      <td>43</td>\n",
       "      <td>39</td>\n",
       "      <td>38</td>\n",
       "      <td>40</td>\n",
       "      <td>39</td>\n",
       "      <td>40</td>\n",
       "      <td>36</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SARS coronavirus strain MA-15, complete genome</th>\n",
       "      <td>6001</td>\n",
       "      <td>6000</td>\n",
       "      <td>6001</td>\n",
       "      <td>6000</td>\n",
       "      <td>6001</td>\n",
       "      <td>6001</td>\n",
       "      <td>6001</td>\n",
       "      <td>6001</td>\n",
       "      <td>6001</td>\n",
       "      <td>6001</td>\n",
       "      <td>...</td>\n",
       "      <td>50</td>\n",
       "      <td>31</td>\n",
       "      <td>27</td>\n",
       "      <td>26</td>\n",
       "      <td>28</td>\n",
       "      <td>27</td>\n",
       "      <td>28</td>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>28 rows  28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     SARS-CoV-2/human/USA/WA-NH10/2020, complete genome  \\\n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome                                                  2     \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome                                                  0     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...                                               6027     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...                                               6025     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...                                               6089     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...                                               6022     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...                                               6028     \n",
       "SARS coronavirus MA15 ExoN1, complete genome                                                     6018     \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...                                               6013     \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...                                               6013     \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...                                               6014     \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...                                               6014     \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...                                               6015     \n",
       "SARS coronavirus MA15, complete genome                                                           6011     \n",
       "SARS coronavirus TW1, complete genome                                                            5993     \n",
       "SARS coronavirus strain MA-15, complete genome                                                   6001     \n",
       "\n",
       "                                                     SARS-CoV-2/human/USA/WA-NH11/2020, complete genome  \\\n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome                                                  1     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...                                               6026     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...                                               6024     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...                                               6088     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...                                               6021     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...                                               6027     \n",
       "SARS coronavirus MA15 ExoN1, complete genome                                                     6017     \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...                                               6012     \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...                                               6012     \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...                                               6013     \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...                                               6013     \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...                                               6014     \n",
       "SARS coronavirus MA15, complete genome                                                           6010     \n",
       "SARS coronavirus TW1, complete genome                                                            5992     \n",
       "SARS coronavirus strain MA-15, complete genome                                                   6000     \n",
       "\n",
       "                                                     SARS-CoV-2/human/USA/WA-NH12/2020, complete genome  \\\n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome                                                  2     \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome                                                  0     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...                                               6027     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...                                               6025     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...                                               6089     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...                                               6022     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...                                               6028     \n",
       "SARS coronavirus MA15 ExoN1, complete genome                                                     6018     \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...                                               6013     \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...                                               6013     \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...                                               6014     \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...                                               6014     \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...                                               6015     \n",
       "SARS coronavirus MA15, complete genome                                                           6011     \n",
       "SARS coronavirus TW1, complete genome                                                            5993     \n",
       "SARS coronavirus strain MA-15, complete genome                                                   6001     \n",
       "\n",
       "                                                     SARS-CoV-2/human/USA/WA-NH14/2020, complete genome  \\\n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome                                                  1     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...                                               6026     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...                                               6024     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...                                               6088     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...                                               6021     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...                                               6027     \n",
       "SARS coronavirus MA15 ExoN1, complete genome                                                     6017     \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...                                               6012     \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...                                               6012     \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...                                               6013     \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...                                               6013     \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...                                               6014     \n",
       "SARS coronavirus MA15, complete genome                                                           6010     \n",
       "SARS coronavirus TW1, complete genome                                                            5992     \n",
       "SARS coronavirus strain MA-15, complete genome                                                   6000     \n",
       "\n",
       "                                                     SARS-CoV-2/human/USA/WA-NH17/2020, complete genome  \\\n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome                                                  2     \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome                                                  0     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...                                               6027     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...                                               6025     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...                                               6089     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...                                               6022     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...                                               6028     \n",
       "SARS coronavirus MA15 ExoN1, complete genome                                                     6018     \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...                                               6013     \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...                                               6013     \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...                                               6014     \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...                                               6014     \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...                                               6015     \n",
       "SARS coronavirus MA15, complete genome                                                           6011     \n",
       "SARS coronavirus TW1, complete genome                                                            5993     \n",
       "SARS coronavirus strain MA-15, complete genome                                                   6001     \n",
       "\n",
       "                                                     SARS-CoV-2/human/USA/WA-NH18/2020, complete genome  \\\n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome                                                  2     \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome                                                  1     \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome                                                  0     \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome                                                  0     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...                                               6027     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...                                               6025     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...                                               6089     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...                                               6022     \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...                                               6028     \n",
       "SARS coronavirus MA15 ExoN1, complete genome                                                     6018     \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...                                               6013     \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...                                               6013     \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...                                               6014     \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...                                               6014     \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...                                               6015     \n",
       "SARS coronavirus MA15, complete genome                                                           6011     \n",
       "SARS coronavirus TW1, complete genome                                                            5993     \n",
       "SARS coronavirus strain MA-15, complete genome                                                   6001     \n",
       "\n",
       "                                                     SARS-CoV-2/human/USA/WA-NH2/2020, complete genome  \\\n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...                                                  1    \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...                                                  1    \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome                                                  2    \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome                                                  1    \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome                                                  1    \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome                                                  0    \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...                                               6027    \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...                                               6025    \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...                                               6089    \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...                                               6022    \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...                                               6028    \n",
       "SARS coronavirus MA15 ExoN1, complete genome                                                     6018    \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...                                               6013    \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...                                               6013    \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...                                               6014    \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...                                               6014    \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...                                               6015    \n",
       "SARS coronavirus MA15, complete genome                                                           6011    \n",
       "SARS coronavirus TW1, complete genome                                                            5993    \n",
       "SARS coronavirus strain MA-15, complete genome                                                   6001    \n",
       "\n",
       "                                                     SARS-CoV-2/human/USA/WA-NH3/2020, complete genome  \\\n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...                                                  1    \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...                                                  1    \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome                                                  2    \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome                                                  1    \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome                                                  1    \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome                                                  0    \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...                                               6027    \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...                                               6025    \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...                                               6089    \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...                                               6022    \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...                                               6028    \n",
       "SARS coronavirus MA15 ExoN1, complete genome                                                     6018    \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...                                               6013    \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...                                               6013    \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...                                               6014    \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...                                               6014    \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...                                               6015    \n",
       "SARS coronavirus MA15, complete genome                                                           6011    \n",
       "SARS coronavirus TW1, complete genome                                                            5993    \n",
       "SARS coronavirus strain MA-15, complete genome                                                   6001    \n",
       "\n",
       "                                                     SARS-CoV-2/human/USA/WA-NH4/2020, complete genome  \\\n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...                                                  1    \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...                                                  1    \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome                                                  2    \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome                                                  1    \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome                                                  1    \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome                                                  0    \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...                                               6027    \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...                                               6025    \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...                                               6089    \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...                                               6022    \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...                                               6028    \n",
       "SARS coronavirus MA15 ExoN1, complete genome                                                     6018    \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...                                               6013    \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...                                               6013    \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...                                               6014    \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...                                               6014    \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...                                               6015    \n",
       "SARS coronavirus MA15, complete genome                                                           6011    \n",
       "SARS coronavirus TW1, complete genome                                                            5993    \n",
       "SARS coronavirus strain MA-15, complete genome                                                   6001    \n",
       "\n",
       "                                                     SARS-CoV-2/human/USA/WA-NH5/2020, complete genome  \\\n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...                                                  2    \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...                                                  1    \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...                                                  2    \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...                                                  1    \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...                                                  2    \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...                                                  2    \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome                                                  2    \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome                                                  2    \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome                                                  2    \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome                                                  0    \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome                                                  1    \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome                                                  1    \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome                                                  2    \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome                                                  2    \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...                                               6027    \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...                                               6025    \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...                                               6089    \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...                                               6022    \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...                                               6028    \n",
       "SARS coronavirus MA15 ExoN1, complete genome                                                     6018    \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...                                               6013    \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...                                               6013    \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...                                               6014    \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...                                               6014    \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...                                               6015    \n",
       "SARS coronavirus MA15, complete genome                                                           6011    \n",
       "SARS coronavirus TW1, complete genome                                                            5993    \n",
       "SARS coronavirus strain MA-15, complete genome                                                   6001    \n",
       "\n",
       "                                                    ...  \\\n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...  ...   \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...  ...   \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...  ...   \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...  ...   \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...  ...   \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...  ...   \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome  ...   \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome  ...   \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome  ...   \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome  ...   \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome  ...   \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome  ...   \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome  ...   \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome  ...   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...  ...   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...  ...   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...  ...   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...  ...   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...  ...   \n",
       "SARS coronavirus MA15 ExoN1, complete genome        ...   \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...  ...   \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...  ...   \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...  ...   \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...  ...   \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...  ...   \n",
       "SARS coronavirus MA15, complete genome              ...   \n",
       "SARS coronavirus TW1, complete genome               ...   \n",
       "SARS coronavirus strain MA-15, complete genome      ...   \n",
       "\n",
       "                                                    SARS coronavirus MA15 ExoN1 isolate P3pp6, complete genome  \\\n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...                                               6028            \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...                                               6027            \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...                                               6028            \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...                                               6027            \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...                                               6028            \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...                                               6028            \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome                                               6028            \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome                                               6028            \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome                                               6028            \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome                                               6028            \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome                                               6027            \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome                                               6027            \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome                                               6028            \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome                                               6028            \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...                                                 20            \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...                                                 16            \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...                                                106            \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...                                                 22            \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...                                                  0            \n",
       "SARS coronavirus MA15 ExoN1, complete genome                                                       25            \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...                                                 29            \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...                                                 28            \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...                                                 30            \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...                                                 29            \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...                                                 30            \n",
       "SARS coronavirus MA15, complete genome                                                             26            \n",
       "SARS coronavirus TW1, complete genome                                                              62            \n",
       "SARS coronavirus strain MA-15, complete genome                                                     50            \n",
       "\n",
       "                                                    SARS coronavirus MA15 ExoN1, complete genome  \\\n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...                                          6018   \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...                                          6017   \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...                                          6018   \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...                                          6017   \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...                                          6018   \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...                                          6018   \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome                                          6018   \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome                                          6018   \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome                                          6018   \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome                                          6018   \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome                                          6017   \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome                                          6017   \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome                                          6018   \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome                                          6018   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...                                            21   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...                                            17   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...                                            85   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...                                            23   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...                                            25   \n",
       "SARS coronavirus MA15 ExoN1, complete genome                                                   0   \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...                                            10   \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...                                             9   \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...                                            11   \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...                                            10   \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...                                            11   \n",
       "SARS coronavirus MA15, complete genome                                                         7   \n",
       "SARS coronavirus TW1, complete genome                                                         43   \n",
       "SARS coronavirus strain MA-15, complete genome                                                31   \n",
       "\n",
       "                                                    SARS coronavirus MA15 isolate P3pp3, complete genome  \\\n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...                                               6012      \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...                                               6012      \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome                                               6012      \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome                                               6012      \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome                                               6013      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...                                                 25      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...                                                 21      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...                                                 91      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...                                                 27      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...                                                 29      \n",
       "SARS coronavirus MA15 ExoN1, complete genome                                                       10      \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...                                                  0      \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...                                                  1      \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...                                                  3      \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...                                                  2      \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...                                                  3      \n",
       "SARS coronavirus MA15, complete genome                                                              3      \n",
       "SARS coronavirus TW1, complete genome                                                              39      \n",
       "SARS coronavirus strain MA-15, complete genome                                                     27      \n",
       "\n",
       "                                                    SARS coronavirus MA15 isolate P3pp4, complete genome  \\\n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...                                               6012      \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...                                               6012      \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome                                               6012      \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome                                               6012      \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome                                               6013      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...                                                 24      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...                                                 20      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...                                                 90      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...                                                 26      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...                                                 28      \n",
       "SARS coronavirus MA15 ExoN1, complete genome                                                        9      \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...                                                  1      \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...                                                  0      \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...                                                  2      \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...                                                  1      \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...                                                  2      \n",
       "SARS coronavirus MA15, complete genome                                                              2      \n",
       "SARS coronavirus TW1, complete genome                                                              38      \n",
       "SARS coronavirus strain MA-15, complete genome                                                     26      \n",
       "\n",
       "                                                    SARS coronavirus MA15 isolate P3pp5, complete genome  \\\n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome                                               6014      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...                                                 26      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...                                                 22      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...                                                 92      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...                                                 28      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...                                                 30      \n",
       "SARS coronavirus MA15 ExoN1, complete genome                                                       11      \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...                                                  3      \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...                                                  2      \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...                                                  0      \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...                                                  3      \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...                                                  4      \n",
       "SARS coronavirus MA15, complete genome                                                              4      \n",
       "SARS coronavirus TW1, complete genome                                                              40      \n",
       "SARS coronavirus strain MA-15, complete genome                                                     28      \n",
       "\n",
       "                                                    SARS coronavirus MA15 isolate P3pp6, complete genome  \\\n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome                                               6013      \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome                                               6014      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...                                                 25      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...                                                 21      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...                                                 91      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...                                                 27      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...                                                 29      \n",
       "SARS coronavirus MA15 ExoN1, complete genome                                                       10      \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...                                                  2      \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...                                                  1      \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...                                                  3      \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...                                                  0      \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...                                                  3      \n",
       "SARS coronavirus MA15, complete genome                                                              3      \n",
       "SARS coronavirus TW1, complete genome                                                              39      \n",
       "SARS coronavirus strain MA-15, complete genome                                                     27      \n",
       "\n",
       "                                                    SARS coronavirus MA15 isolate P3pp7, complete genome  \\\n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...                                               6015      \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...                                               6015      \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...                                               6015      \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...                                               6015      \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome                                               6015      \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome                                               6015      \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome                                               6015      \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome                                               6015      \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome                                               6014      \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome                                               6015      \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome                                               6015      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...                                                 26      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...                                                 22      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...                                                 92      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...                                                 28      \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...                                                 30      \n",
       "SARS coronavirus MA15 ExoN1, complete genome                                                       11      \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...                                                  3      \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...                                                  2      \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...                                                  4      \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...                                                  3      \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...                                                  0      \n",
       "SARS coronavirus MA15, complete genome                                                              4      \n",
       "SARS coronavirus TW1, complete genome                                                              40      \n",
       "SARS coronavirus strain MA-15, complete genome                                                     28      \n",
       "\n",
       "                                                    SARS coronavirus MA15, complete genome  \\\n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...                                    6011   \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...                                    6010   \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...                                    6011   \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...                                    6010   \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...                                    6011   \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...                                    6011   \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome                                    6011   \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome                                    6011   \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome                                    6011   \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome                                    6011   \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome                                    6010   \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome                                    6010   \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome                                    6011   \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome                                    6011   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...                                      22   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...                                      18   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...                                      88   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...                                      24   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...                                      26   \n",
       "SARS coronavirus MA15 ExoN1, complete genome                                             7   \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...                                       3   \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...                                       2   \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...                                       4   \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...                                       3   \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...                                       4   \n",
       "SARS coronavirus MA15, complete genome                                                   0   \n",
       "SARS coronavirus TW1, complete genome                                                   36   \n",
       "SARS coronavirus strain MA-15, complete genome                                          24   \n",
       "\n",
       "                                                    SARS coronavirus TW1, complete genome  \\\n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...                                   5993   \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...                                   5992   \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...                                   5993   \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...                                   5992   \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...                                   5993   \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...                                   5993   \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome                                   5993   \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome                                   5993   \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome                                   5993   \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome                                   5993   \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome                                   5992   \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome                                   5992   \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome                                   5993   \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome                                   5993   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...                                     58   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...                                     54   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...                                    124   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...                                     60   \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...                                     62   \n",
       "SARS coronavirus MA15 ExoN1, complete genome                                           43   \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...                                     39   \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...                                     38   \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...                                     40   \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...                                     39   \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...                                     40   \n",
       "SARS coronavirus MA15, complete genome                                                 36   \n",
       "SARS coronavirus TW1, complete genome                                                   0   \n",
       "SARS coronavirus strain MA-15, complete genome                                         16   \n",
       "\n",
       "                                                    SARS coronavirus strain MA-15, complete genome  \n",
       " SARS-CoV-2/human/USA/WA-NH10/2020, complete ge...                                            6001  \n",
       " SARS-CoV-2/human/USA/WA-NH11/2020, complete ge...                                            6000  \n",
       " SARS-CoV-2/human/USA/WA-NH12/2020, complete ge...                                            6001  \n",
       " SARS-CoV-2/human/USA/WA-NH14/2020, complete ge...                                            6000  \n",
       " SARS-CoV-2/human/USA/WA-NH17/2020, complete ge...                                            6001  \n",
       " SARS-CoV-2/human/USA/WA-NH18/2020, complete ge...                                            6001  \n",
       " SARS-CoV-2/human/USA/WA-NH2/2020, complete genome                                            6001  \n",
       " SARS-CoV-2/human/USA/WA-NH3/2020, complete genome                                            6001  \n",
       " SARS-CoV-2/human/USA/WA-NH4/2020, complete genome                                            6001  \n",
       " SARS-CoV-2/human/USA/WA-NH5/2020, complete genome                                            6001  \n",
       " SARS-CoV-2/human/USA/WA-NH6/2020, complete genome                                            6000  \n",
       " SARS-CoV-2/human/USA/WA-NH7/2020, complete genome                                            6000  \n",
       " SARS-CoV-2/human/USA/WA-NH8/2020, complete genome                                            6001  \n",
       " SARS-CoV-2/human/USA/WA-NH9/2020, complete genome                                            6001  \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp10, com...                                              46  \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp3, comp...                                              42  \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp4, comp...                                             112  \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp5, comp...                                              48  \n",
       "SARS coronavirus MA15 ExoN1 isolate P3pp6, comp...                                              50  \n",
       "SARS coronavirus MA15 ExoN1, complete genome                                                    31  \n",
       "SARS coronavirus MA15 isolate P3pp3, complete g...                                              27  \n",
       "SARS coronavirus MA15 isolate P3pp4, complete g...                                              26  \n",
       "SARS coronavirus MA15 isolate P3pp5, complete g...                                              28  \n",
       "SARS coronavirus MA15 isolate P3pp6, complete g...                                              27  \n",
       "SARS coronavirus MA15 isolate P3pp7, complete g...                                              28  \n",
       "SARS coronavirus MA15, complete genome                                                          24  \n",
       "SARS coronavirus TW1, complete genome                                                           16  \n",
       "SARS coronavirus strain MA-15, complete genome                                                   0  \n",
       "\n",
       "[28 rows x 28 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "distances"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Phylo.TreeConstruction import DistanceMatrix\n",
    "import numpy as np \n",
    "\n",
    "matrix = np.tril(distances.values).tolist()\n",
    "for i in range(len(matrix)):\n",
    "    matrix[i] = matrix[i][:i+1]\n",
    "dm = DistanceMatrix(list(distances.index), matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Phylo.TreeConstruction import DistanceTreeConstructor\n",
    "\n",
    "constructor = DistanceTreeConstructor()\n",
    "\n",
    "# tree = constructor.nj(dm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'tree' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-38-f868b668027e>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mBio\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mPhylo\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m \u001b[0mtree\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mladderize\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m   \u001b[0;31m# Flip branches so deeper clades are displayed at top\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      5\u001b[0m \u001b[0mPhylo\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdraw\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtree\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'tree' is not defined"
     ]
    }
   ],
   "source": [
    "%matplotlib inline\n",
    "from Bio import Phylo\n",
    "\n",
    "tree.ladderize()   # Flip branches so deeper clades are displayed at top\n",
    "Phylo.draw(tree)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
